### PR TITLE
Mention the importance of closing an allocated instance and improve migration notes

### PR DESCRIPTION
**Pull request by remeh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,15 +24,36 @@ require 'datadog/statsd'
...
 
+### Migrating from v4.x to v5.x
+
+If you are already using DogStatsD-ruby v4.x and you want to migrate to a version v5.x, the major
+change concerning you is the new threading model (please see section Threading model):
+
+In practice, it means two things:
+
+1. Now that the client is buffering metrics before sending them, you have to manually
+call the method `Datadog::Statsd#flush` if you want the metrics to be sent. Note that the companion thread will automatically flush the buffered metrics if the buffer gets full or when you are closing the instance.
```
Comment (kbogtob) [inline README.md:50]: `if you want to force the sending of metrics`?

File: README.md [hunk 2]
```diff
@@ -24,15 +24,36 @@ require 'datadog/statsd'
...
+
+2. You have to make sure you are either:
+
+  * using singletons instances of the DogStatsD client and not allocating one each time you need one, or,
```
Comment (kbogtob) [inline README.md:54]: `... and letting the buffering mechanism flush metrics`?

File: README.md [hunk 3]
```diff
@@ -90,7 +111,9 @@ On versions greater than 5.0, we changed the threading model of the library so t
 
 When you instantiate a `Datadog::Statsd`, a companion thread is spawned. This thread will be called the Sender thread, as it is modeled by the [Sender](../lib/datadog/statsd/sender.rb) class.
 
-This thread is automatically stopped when you close the statsd client (`Datadog::Statsd#close`). The communication between the current thread is managed through a standard Ruby Queue.
+This thread is stopped when you close the statsd client (`Datadog::Statsd#close`). It also means that allocating a lot of statsd clients without closing them properly when not used anymore
+could lead to a thread leak.
```
Comment (kbogtob) [inline README.md:115]: (even though they will be sleeping, blocked on IO)

**Review by kbogtob — APPROVED** [review]
That's good explanations! I left some changes proposals.

File: CHANGELOG.md [hunk 4]
```diff
@@ -38,9 +54,11 @@ statsd.gauge('example_metric.gauge', 123, tags: ['environment:dev'])
 
 # synchronous flush
 statsd.flush(sync: true)
-```
 
-2. `Statsd#initialize` parameter `max_buffer_bytes` has been renamed to `buffer_max_payload_size` for consistency with the new automatic batch strategy. Please note the addition of `buffer_max_pool_size` to limit the maximum amount of *messages* to buffer.
+...
+
+statsd.close()
+```
```
[the remark this answers is not in the document]
Comment (remeh) [reply CHANGELOG.md:61]: I would avoid it since we are deprecating it.

**Pull request merged**